python/lsst/ip/isr/isrMockLSST.py

```diff
@@ -0,0 +1,597 @@
+                                       0., self.config.readNoise / self.config.gain)
+
+        # 4. Apply cross-talk in ADU
+        if self.config.doAddCrosstalk:
+            ctCalib = CrosstalkCalib()
+            for idxS, ampS in enumerate(exposure.getDetector()):
+                for idxT, ampT in enumerate(exposure.getDetector()):
+                    ampDataT = exposure.image[ampT.getRawDataBBox().shiftedBy(ampT.getRawXYOffset())]
+                    outAmp = ctCalib.extractAmp(exposure.getImage(), ampS, ampT,
+                                                isTrimmed=False)
+                    self.amplifierAddCT(outAmp, ampDataT, self.crosstalkCoeffs[idxS][idxT])
+
+        for amp in exposure.getDetector():
+            # Get image bbox and data
+            imageBBox = amp.getRawDataBBox().shiftedBy(amp.getRawXYOffset())
+            ampData = exposure.image[imageBBox]
+
+            # Get overscan bbox and data
+            parallelOscanBBox = amp.getRawParallelOverscanBBox().shiftedBy(amp.getRawXYOffset())
+            parallelOscanData = exposure.image[parallelOscanBBox]
+
+            serialOscanBBox = amp.getRawSerialOverscanBBox().shiftedBy(amp.getRawXYOffset())
+
+            # 5. Apply parallel overscan in ADU
+            if self.config.doAddParallelOverscan:
+                # Add noise to the parallel overscan region.
+                self.amplifierAddNoise(parallelOscanData, 0.0,
+                                       self.config.readNoise / self.config.gain)
+
+                # # Apply gradient along Y axis to the image
+                # # and parallel overscan regions.
+                self.amplifierAddYGradient(ampData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+                self.amplifierAddYGradient(parallelOscanData, -1.0 * self.config.overscanScale,
+                                           1.0 * self.config.overscanScale)
+
+        # 6. Add Parallel overscan xtalk.
+        # TODO: DM-43286
+
+            if self.config.doAddSerialOverscan:
+                # 1. We grow the image to the parallel oversan region
+                # (we do this instead of using the whole raw region
+                # in case there are prescan regions)
+                grownImageBBox = imageBBox.expandedTo(parallelOscanBBox)
+                # 2. Now we grow the serial overscan region
+                # to include the corners
+                serialOscanBBox = geom.Box2I(
+                    geom.Point2I(serialOscanBBox.getMinX(),
+                                 grownImageBBox.getMinY()),
+                    geom.Extent2I(serialOscanBBox.getWidth(),
+                                  grownImageBBox.getHeight()),
+                )
+                serialOscanData = exposure.image[serialOscanBBox]
+
+                # 7. Add bias level to the whole region of amplifiers in ADU
+                self.amplifierAddNoise(ampData, self.config.biasLevel,
```

Comment:
This (combined with the above call with zero mean) has double-added the read noise, at least in the imaging region.  The bias signal contributes to the image by adding:

- A DC offset (`self.config.biasLevel`) which has a known dispersion (`self.config.readNoise`)
- Two 1-D contributions (the serial and parallel overscan signals)
- A final 2-D residual pattern (what the combined bias will correct).

The `doAddBias` step above should now only be adding this third residual pattern, as the read noise and bias level additions have been moved here to ensure that the two overscan regions are consistent.  If we're not adding a residual pattern, that step should add 0.0 there.

I was confused on units at this point, but `self.config.readNoise` claims to be in electrons, the comments here indicate we're working in ADU, so the factor of `1 / self.config.gain` is correct, which I'm stating here in case I get confused again in the future.